src/zui/ZUICreatePerson/checkInvalidFields.ts

```diff
@@ -10,6 +10,19 @@ export default function checkInvalidFields(
 ) {
   let invalidFields: string[] = [];
 
+  //name
+  if (personalInfo.first_name === '') {
+    invalidFields.push('first_name');
+  } else {
+    invalidFields = invalidFields.filter((item) => item !== 'first_name');
+  }
```

Comment:
I understand that you've followed a pre-existing pattern here, so this isn't so much commenting on your work as it is a comment on the resulting code. But I don't understand what the purpose of these `else` blocks are. When this code is reached, the `invalidFields` array is guaranteed to be empty, because it was created empty on line 11, not modified on line 15 (because that line was never reached) and then we arrive here.

I wonder if these filterings are remnants of some point where the `invalidFields` was stored persistently across renders or something? Because as far as I can understand from just looking at this, all else statements in this function are redundant, and `invalidFields` could be a `const`.

Reply:
(The phone number bug is already filed: https://github.com/zetkin/app.zetkin.org/issues/2019)